enhancements/platforms/ovirt-rhv-ipi.md

```diff
@@ -0,0 +1,271 @@
+if software-defining your datacenter, utilizing existing hardware and making \ 
+that pain free and fast.
+Like `baremetal` platform it uses the [OpenShift-Metal³ kni-installer](https://github.com/openshift-metal3/kni-installer) method \
+- essentially providing an internal cluster-level DNS service using mDNS and
+ coreDNS, and load-balancing using keepalived.
+
+Components involved:
+  - github.com/openshift/api
+  - github.com/openshift/installer
+  - github.com/openshift/machine-config-operator
+  - github.com/openshift/cluster-image-registry-operator (@adambkapln?)
+  - github.com/ovirt/cluster-api-provider-ovirt
+  - github.com/ovirt/terraform-provider-ovirt
+
+## Motivation
+
+- It has been defined as a key initiative for 4.3
+- The feedback RHV team got is that there is a lot of demand for this 
+  kind of installation
+
+### Goals
+
+- provide a way to install Openshift on oVirt infrastructure using 
+  the installer - an IPI installation.
+- implementing a cluster-api provider to provide scaling and managing the cluster
+  nodes (used by IPI, and useful for UPI, and also node management/fencing)
+
+### Non-Goals
+- UPI implementation will be provided separately.
+
+## Proposal
+
+This provider enables the Openshift Installer to provision VM resources in an \
+oVirt data center, that will be used as worker and masters of the clusters. It \
+will also create the bootstrap machine, and the configuration needed to get \
+the initial cluster running by supplying DNS a service and load balancing, all \
+using static pods.
+
+This work is related to the Bare-Metal provider because oVirt does not supply \
+DNS and LB services but is a platform provider. See the [baremetal ipi networking doc][baremetal-ipi-networking]
+
+### Implementation Details/Notes/Constraints [optional]
+
+1. Survey
+
+The installation starts and right after the user supplies his public ssh key,\
+and then choose `ovirt` the installation will ask for all the relevant details\
+of the installation: **url**, **user** and **password** of ovirt api. The installer \
+will validate it can communicate with the api, otherwise it will fail to proceed.\
+Next the installer will ask to choose the ovirt **cluster**, where the VMs will be\
+created, and next would be a *template* from that cluster for those VMs.\
+*Note* that it is the user's responsibility to upload the relevant RHCOS template\
+to ovirt. This may change in the future.
+With that the survery continues to the general cluster name, domain name, and\
+the rest of the non-ovirt specific question.
+
+2. Resource creation - terraform
+
+All the information from the survey is serialized by the installer into the file\ `terraform.ovirt.auto.tfvars.json` under the current directory and then it invokes\
+terraform, which automatically load `.auto.tfvars.json`. `terraform.tfstate` is\
+the actual declaration of the resources and with it terraform creates the VMs.\
+
+3. Bootstrap
```

Comment:
This needs to take into account how cluser-etcd-operator fits into bootstrapping.

Reply:
Can you elaborate?

Reply:
https://github.com/openshift/enhancements/pull/56

Reply:
The impact of the etcd-operator on the need publishing of the SRV records will probably mean that ovirt, baremetal and openstack platforms can have simpler bootstraping,  but it is not yet understood how this is going to change. I'll add it as related enhancement.